File: spec-n-roll/src/spec-n-roll-runtime/src/application/output/runtime-output-writer.ts
/**
 * Writes user-facing runtime output.
 */
export interface RuntimeOutputWriter {
  /**
   * Writes a line of output to the runtime console.
   *
   * @param text - Text to write. The value should not include a trailing newline.
   */
  writeLine(text: string): void;
}

File: atlas/src/tools/atlas-rb/lib/atlas/rb/runtime_output_writer.rb
# frozen_string_literal: true

module Atlas
  module Rb
    # Owns the generator's deliberate user-facing stdout and stderr contract.
    class RuntimeOutputWriter
      # Creates output from explicit writable streams.
      def initialize(output_stream, error_stream)
        @output_stream = output_stream
        @error_stream = error_stream
      end

      # Writes one successful command result.
      def write_line(message)
        @output_stream.puts(message)
      end

      # Writes one user-facing command failure.
      def write_error(message)
        @error_stream.puts(message)
      end
    end
  end
end
